File: connectors/connector-mantis/src/main/java/com/taskadapter/connector/mantis/MantisProjectConverter.java
package com.taskadapter.connector.mantis;

import biz.futureware.mantis.rpc.soap.client.ProjectData;
import com.taskadapter.model.GProject;

import java.util.List;
import java.util.stream.Collectors;

public class MantisProjectConverter {
    public static List<GProject> toGProjects(List<ProjectData> objects) {
        return objects.stream().map(MantisProjectConverter::toGProject)
                .collect(Collectors.toList());
    }

    private static GProject toGProject(ProjectData mantisProject) {
        return new GProject()
                .setId(mantisProject.getId().longValue())
                .setName(mantisProject.getName())
                .setKey(String.valueOf(mantisProject.getId()))
                .setDescription(mantisProject.getDescription())
                .setHomepage("");
    }
}

File: connectors/connector-mantis/src/main/scala/com/taskadapter/connector/mantis/MantisProjectConverter.scala
package com.taskadapter.connector.mantis

import java.util

import biz.futureware.mantis.rpc.soap.client.ProjectData
import com.taskadapter.model.GProject

class MantisProjectConverter {
  def toGProjects(objects: util.List[ProjectData]): util.List[GProject] = {
    val projects = new util.ArrayList[GProject]
    import scala.collection.JavaConversions._
    for (rmProject <- objects) {
      val project = toGProject(rmProject)
      projects.add(project)
    }
    projects
  }

  def toGProject(mantisProject: ProjectData) = new GProject()
    .setId(mantisProject.getId.longValue())
    .setName(mantisProject.getName)
    .setKey(String.valueOf(mantisProject.getId))
    .setDescription(mantisProject.getDescription)
    .setHomepage("")
}
